src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Renamer.cs

```diff
@@ -76,101 +71,15 @@ internal class Renamer
-                if (declaration == null)
-                    return renamedSolution;
-
-                var semanticModel = await newDocument.GetSemanticModelAsync().ConfigureAwait(false);
-                if (semanticModel == null)
-                    return renamedSolution;
-
-                var symbol = semanticModel.GetDeclaredSymbol(declaration);
-                if (symbol == null)
-                    return renamedSolution;
-
-                bool userConfirmed = await CheckUserConfirmation().ConfigureAwait(false);
-                if (!userConfirmed)
-                    return renamedSolution;
-
-                string newName = Path.GetFileNameWithoutExtension(newDocument.FilePath);
-                
-                // Note that RenameSymbolAsync will return a new snapshot of solution.
-                renamedSolution = await _roslynServices.RenameSymbolAsync(newDocument.Project.Solution, symbol, newName).ConfigureAwait(false);
-                project = renamedSolution.Projects.Where(p => StringComparers.Paths.Equals(p.FilePath, myNewProject.FilePath)).FirstOrDefault();
-            }
-            return null;
-        }
-
-        private async Task<bool> CheckUserConfirmation()
-        {
-            if (_userPromptedOnce)
-            {
-                return _userConfirmedRename;
-            }
-
-            await _threadingService.SwitchToUIThread();
-            var userNeedPrompt = _optionsSettings.GetPropertiesValue("Environment", "ProjectsAndSolution", "PromptForRenameSymbol", false);
-            if (userNeedPrompt)
-            {
-                string renamePromptMessage = string.Format(CultureInfo.CurrentCulture, Resources.RenameSymbolPrompt, Path.GetFileNameWithoutExtension(_oldFilePath));
-
-                await _threadingService.SwitchToUIThread();
-                _userConfirmedRename = _userNotificationServices.Confirm(renamePromptMessage);
-            }
-
-            _userPromptedOnce = true;
-            return _userConfirmedRename;
-        }
+        public async Task RenameAsync(Project project) =>
+            await (GetStrategy()?.RenameAsync(project, _oldFilePath, _newFilePath) ?? Task.CompletedTask).ConfigureAwait(false);
 
-        private bool HasMatchingSyntaxNode(Document document, SyntaxNode syntaxNode, string name)
+        private IRenameStrategy GetStrategy()
         {
-            var generator = SyntaxGenerator.GetGenerator(document);
-            var kind = generator.GetDeclarationKind(syntaxNode);
-
-            if (kind == DeclarationKind.Class ||
-                kind == DeclarationKind.Interface ||
-                kind == DeclarationKind.Delegate ||
-                kind == DeclarationKind.Enum ||
-                kind == DeclarationKind.Struct)
-            {
-                return generator.GetName(syntaxNode) == name;
-            }
-            return false;
+            IRenameStrategy[] strategies = new IRenameStrategy[] {
```

Comment:
Long term we should make the "Renamer" a service that that imports other renamers.